posthog/api/action.py

```diff
@@ -271,73 +270,88 @@ def _stickiness(self, action: Action, filters: Dict[Any, Any], request: request.
             'data': data
         }
 
-    def _serialize_action(self, action: Action, filters: Dict[Any, Any], request: request.Request) -> Dict:
-        append = {
+    def _serialize_entity(self, id: str, name: str, filtered_events: QuerySet, filters: Dict[Any, Any], request: request.Request) -> Dict:
+        entity: Dict[str, any] = {
             'action': {
-                'id': action.pk,
-                'name': action.name
+                'id': id,
+                'name': name
             },
-            'label': action.name,
+            'label': name,
             'count': 0,
             'breakdown': []
         }
+
         if request.GET.get('shown_as', 'Volume') == 'Volume':
-            append.update(self._aggregate_by_day(action=action, filters=filters, request=request))
+            entity.update(self._aggregate_by_day(filtered_events=filtered_events, filters=filters, request=request))
         elif request.GET['shown_as'] == 'Stickiness':
-            append.update(self._stickiness(action=action, filters=filters, request=request))
-        return append
+            entity.update(self._stickiness(filtered_events=filtered_events, filters=filters, request=request))
+        return entity
 
-    def _serialize_people(self, action: Action, people: QuerySet, request: request.Request) -> Dict:
+    def _serialize_people(self, id: str, name: str, people: QuerySet, request: request.Request) -> Dict:
         people_dict = [PersonSerializer(person, context={'request': request}).data for person in  people]
         return {
             'action': {
-                'id': action.pk,
-                'name': action.name
+                'id': id,
+                'name': name
             },
             'people': people_dict,
             'count': len(people_dict)
         }
 
+    def _process_entity_for_events(self, entity, entity_type=None):
+        if entity_type == 'action':
+            filtered_events = Event.objects.filter_by_action(action=entity)
+            return filtered_events
+        elif entity_type == 'event':
+            filtered_events = Event.objects.filter(event=entity['id'], team=self.request.user.team_set.get())
+            return filtered_events
+        return None
+
     @action(methods=['GET'], detail=False)
     def trends(self, request: request.Request, *args: Any, **kwargs: Any) -> Response:
         actions = self.get_queryset()
         actions = actions.filter(deleted=False)
         actions_list = []
 
-        parsed_actions = self._parse_actions(request)
+        parsed_events = self._parse_entities('events')
 
-        if parsed_actions:
-            for filters in parsed_actions:
```

Comment:
was this necessary? I deleted it because it was a redundant retrieval of the actions in the query param